### PR TITLE
Verify spoiler.xml against Cockatrice's XSD file

### LINKED ISSUE #50: Run XML through Cockatrice's XSD
Cockatrice provides an XSD file that can be used to verify the integrity of an XML file in conforming to the card database standard.  The spoiler XMLs should be verified against it on every update.

https://raw.githubusercontent.com/Cockatrice/Cockatrice/master/doc/cards.xsd

### PR DESCRIPTION
**Pull request by dev-id**
verify_xml currently takes an xml file and XSD as a string

Prints a pass/fail above XML dump

Fixes #50

### CODE REVIEW

**Comment by dev-id** [comment]
Fixes #50

File: main.py [hunk 1]
```diff
@@ -138,6 +147,11 @@ def save_xml(xmlstring, outfile):
     save_setjson(combinedjson, 'spoiler')
     spoilers.write_combined_xml(combinedjson, setinfos)
     save_xml(spoilers.pretty_xml('out/spoiler.xml'), 'out/spoiler.xml')
+    cockatrice_xsd = requests.get('https://raw.githubusercontent.com/Cockatrice/Cockatrice/master/doc/cards.xsd').text
+    if verify_xml('out/spoiler.xml', cockatrice_xsd):  # check if our XML passes Cockatrice's XSD
+        print 'XML passes XSD verification'
+    else:
+        print 'XML fails XSD verification'
```
Comment (tooomm) [inline main.py]: Specify in the message that it's the XSD from Cockatrice?
Also make clear that just "spoiler.xml" is tested.

What happens if location of the xsd file changes or file is not available etc.? Is there an helpful error in that case?

**Comment by tooomm** [comment]
You just verify spoiler.xml because that's the only file we will keep in the long run?

There won't be an useful error if the file doesn't pass?

**Comment by dev-id** [comment]
>  location of the xsd file changes or file is not available etc.

No clue.  We could include it in our repo, but it wouldn't update.

> [only] spoiler.xml

As spoiler.xml contains the other XMLs, if it passes, the others should too.  They haven't had their keys re-ordered like in my second commit here, but the `sequence` requirement from the XSD appears to be wrong as Cockatrice will take the keys/tags in any order.

> useful error

This particular implementation only seems to return true/false.

Edit: I see there is an exception generating variation on the function, I will investigate that.

**Comment by tooomm** [comment]
>This particular implementation only seems to return true/false.

Which is something.
We can decide to not close (+extend) the initial issue and have this as a first step.
If it fails it would be nice to know where and why. Same for the yml input.

If the file isn't available the log should tell us, instead maybe just throwing "fails".
Can you try and change the file link to a different place or other file and see what happens?

<br>
To assume the other set files should pass is not perfect, but ok. :)

**Comment by dev-id** [comment]
Now errors and prints error information but doesn't stop build for:
Bad XSD
Bad XML
Invalid schema

**Pull request merged**